### PR TITLE
Anton/improve api

### PR DESCRIPTION
**Pull request by antonoberg**

### CODE REVIEW

File: packages/api/src/lib/postgres/models/createModel.batch.spec.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+            return model.Model.bulkCreate(batchData).then(res => {
+              return res
+            })
+          }
+
+          return batchExecute({
+            createEntry,
+            execute,
+            numberOfEntries: 300000,
+            numberOfentriesEachBatch: 10000,
```
Comment (fredrikolovsson) [inline packages/api/src/lib/postgres/models/createModel.batch.spec.js]: cool! maybe fix camelCase entries => Entries

File: packages/api/src/lib/postgres/models/createModel.batch.spec.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+// 1k each batch
+// console.log console.js:171
+//   290000.batch.execute: 3241.546ms
+// console.log console.js:171
+//   batch: 85810.160ms
+// console.log src/lib/postgres/models/createModel.batch.spec.js:165
+//   test:batch
+
+// 100k
+// 100k each batch - heap out of memory
```
Comment (fredrikolovsson) [inline packages/api/src/lib/postgres/models/createModel.batch.spec.js]: want to keep those? maybe use `/* ... */` syntax and add a comment then

File: packages/api/src/utilities/test/batchDescribe.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const config = require('../../config')
+
+module.exports = function batchDescribe(testName, ...rest) {
+  if (config.test.runBatchTests) {
+    return describe(testName, ...rest)
+  }
+
+  return describe(testName, () => {
+    it('Does not run when API_TESTS not set to true', () => {
```
Comment (fredrikolovsson) [inline packages/api/src/utilities/test/batchDescribe.js]: API_TESTS => BATCH_TESTS

File: packages/api/src/utilities/test/batchExecute.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+  const batchIdentifier = `${count}.batch`
+  const executeIdentifier = `${batchIdentifier}.execute`
+
+  const batchData = []
+  /* eslint-disable no-plusplus */
+  for (let index = 0; index < numberOfBatchEntries; index++) {
+    batchData[index] = createEntry(count + index)
+  }
+  /* eslint-enabke no-plusplus */
```
Comment (fredrikolovsson) [inline packages/api/src/utilities/test/batchExecute.js]: doesn't really matter but: enabke => enable

(and maybe using `i += 1` should be preferred to disabling the lint rule)

File: packages/api/src/utilities/test/batchExecute.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+  console.time(executeIdentifier)
+  return execute(batchData).then(() => {
+    console.timeEnd(executeIdentifier)
+    return runBatch({
+      count: count + numberOfBatchEntries,
```
Comment (fredrikolovsson) [inline packages/api/src/utilities/test/batchExecute.js]: `numberOfEntriesInThisBatch` would have been more clear to me. Now I really had to think thrice to distinguish the different numbers :-)

File: packages/common/dist/openApi.json [hunk 6]
```diff
@@ -37,6 +37,13 @@
       "description": "Description of collections api\n",
       "name": "collectionsApi"
     },
+    {
+      "name": "statusApi"
+    },
+    {
+      "description": "Description of auth api\n",
+      "name": "statusApu"
```
Comment (fredrikolovsson) [inline packages/common/dist/openApi.json]: rebuild after removing this export (commented before)

File: packages/api/src/utilities/test/waitForApiRestart.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  { operationId = 'getStatus', maxTime = 5000 } = {}
+) {
+  return new Promise((resolve, reject) => {
+    let timedOut = false
+
+    setTimeout(() => {
+      timedOut = true
+    }, maxTime)
+
+    const poll = () => {
```
Comment (fredrikolovsson) [inline packages/api/src/utilities/test/waitForApiRestart.js]: nice

File: packages/api/src/apis/statusApi/index.js [hunk 8]
```diff
@@ -0,0 +1,4 @@
+exports.info = require('./info')
+exports.endpoints = require('./endpoints')
+
+exports.name = 'status'
```
[the remark this answers is not in the document]
Comment (antonoberg) [reply packages/api/src/apis/statusApi/index.js]: Lets get back to this

File: packages/api/src/lib/postgres/models/createModel.batch.spec.js (hunk 1 above)
Comment (antonoberg) [reply packages/api/src/lib/postgres/models/createModel.batch.spec.js]: Fix later

File: packages/api/src/lib/postgres/models/createModel.batch.spec.js (hunk 2 above)
Comment (antonoberg) [reply packages/api/src/lib/postgres/models/createModel.batch.spec.js]: I leave it like this one but will get back to it

File: packages/common/src/endpointFactory/createEndpointFactory.js [hunk 9]
```diff
@@ -54,33 +54,36 @@ const getSchemaFromResponse = response => {
   )
 }
 
-const getBodyValidator = ({ methodSpecification }) => {
+const getBodyValidator = ({ origin, methodSpecification }) => {
```
[the remark this answers is not in the document]
Comment (antonoberg) [reply packages/common/src/endpointFactory/createEndpointFactory.js]: Likely not but will get back to that

File: packages/common/src/error/validators/systemModelSchema.js [hunk 10]
```diff
@@ -1,20 +1,27 @@
...
 module.exports = function createSystemModelSchemaValidator({
+  context,
   dataPath,
   model,
+  origin = 'client',
   schema,
   throwOnError,
 }) {
+  const errorHandler =
+    origin === 'server'
```
[the remark this answers is not in the document]
Comment (antonoberg) [reply packages/common/src/error/validators/systemModelSchema.js]: Yes will get back to that :)

**Pull request merged**